diff -Nuar metis-5.1.0/GKlib/GKlibSystem.cmake.orig metis-5.1.0/GKlib/GKlibSystem.cmake
--- metis-5.1.0/GKlib/GKlibSystem.cmake.orig	2015-10-27 20:17:16.000000000 -0700
+++ metis-5.1.0/GKlib/GKlibSystem.cmake	2015-10-27 20:17:35.000000000 -0700
@@ -33,7 +33,7 @@
       set(GKlib_COPTIONS "${GKlib_COPTIONS} -fPIC")
   endif(NOT MINGW)
 # GCC warnings.
-  set(GKlib_COPTIONS "${GKlib_COPTIONS} -Wall -pedantic -Wno-unused-but-set-variable -Wno-unused-variable -Wno-unknown-pragmas")
+  set(GKlib_COPTIONS "${GKlib_COPTIONS} -Wall -pedantic")
 elseif(${CMAKE_C_COMPILER_ID} MATCHES "Sun")
 # Sun insists on -xc99.
   set(GKlib_COPTIONS "${GKlib_COPTIONS} -xc99")
